src/timer.rs

```diff
@@ -0,0 +1,70 @@
+trait Timer: Sized {
+    fn start(milliseconds: u32) -> Self;
+    fn has_elapsed(self) -> Option<Self>;
+}
+
+use libtock_core::syscalls;
```

Comment:
nit: This should be first in the file. (Given that this file is going to be split into `api::timer` and `env::tock::timer` it's not really an issue.)

Reply:
Done.